### DISCUSSION TITLE
device.clocksynchronizer disables behaviour device

### DISCUSSION THREAD

**Discussion by ablot** [harp-tech/protocol#90] [Q&A]
I'm not exactly sure what my issue is. I have a analog input device and a behaviour device. Both of them work fine but have different harp clock (as expected). I got a clocksynchronizer to solve that but when I connect the clock synchronizer the behaviour device is not detected anymore. 

In bonsai the workflow runs but does not emit anything. In the behaviour GUI, only the analog input appears in the list of connected devices.

I tried to disconnect all the clock jack cables (powering eveything down before doing so) and connect only the usb cables. That already triggered the issue. If I have usb/power cables connected to both the behaviour and analog device everything work. As soon as I connect the clock synchronizer the behaviour device disappears (but the analog device still works).

Is it likely to be a hardware issue? A missing driver? Or is there maybe a bonsai step that I need to run to start the synchroniser which I forgot (and my behaviour device is stuck waiting for a clock signal)?

**Comment by ablot** [comment]
part of it might have been a COM port issue (not being closedor something like that). After rebooting the computer, I can have the 3 device connected to the USB at the same time.

But the issue remains once I connect the clock cables. The analog input device seems to work as expected (sending messages with a harp clock timestamp that has now a low value) while the behaviour device does not emit any messages anymore.

**Comment by artursilva0** [comment]
A couple of questions to try to identify the issue:

1. How are you connecting the clock cables? One cable from one of the clock synchronizer outputs to the behavior and another one to the analog input? Are you connecting the clock cables with the boards powered down (both USB and power) and only after that, powering the boards?
2. When you connect the clock cable to the analog device, the state LEDs of both boards blink simultaneously?
3. When the behavior board "disappears", does the state LED continues to blink?

**Comment by ablot** [comment]
Hi,

1. Yes, one cable from clkout6 to the clkin of the behaviour and one cable clkout4 to the the analog input. The three boards were powered down when I connected the cable. Then I powered them in that order: clock synch, analog, behaviour. Does the order matter?
2. The LED of the clock synchronizer does not blink, only that of the analog device. If I power down everything, remove all clock cables and connect only the USB of the clock synchronizer, its LED blinks at onset (3 bursts of a couple of flashes) but then it goes black
3. No, the LED is black when I cannot detect the behaviour device (and "VisualIndicators" is On in the bonsai workflow).

If I connect only the clock synchronizer, start bonsai and double click on the clock synchroniser device, I never get the connection. The bottom of the window displays "Connecting to device..." forever.

**Comment by ablot** [comment]
A screenshot in case it helps:
![clocksynchnotfound](https://github.com/harp-tech/harp-tech.github.io/assets/2701493/c0120bad-57ba-4d69-8e3b-bd0328c7a6ee)

The bonsai console is also empty.

**Comment by artursilva0** [comment]
Taking into account what you've described it seems that somehow the device firmware got corrupted and/or the device got damaged. 

You can try to update the firmware and see if it works
1. Install the the Harp to CSV GUI [here](https://bitbucket.org/fchampalimaud/downloads/downloads/Harp_Convert_To_CSV_v1.8.3.zip)
2. Under the tab *Options* write "bootloader" in the *List* text box
3. Select the correspondent COM port
4. Select the .hex file (you can download the latest .hex [here](https://github.com/harp-tech/device.clocksynchronizer/releases))
5. If everything goes well you should get no errors and the progress bar will reach the end 

You should now be able to see after the initial LED burst, the LED toggling after a couple of seconds. 

If not, or if the firmware update fails I would suggest you to request an RMA to replace the clock synchronizer board.

**Comment by ablot** [comment]
It seems that there is another issue. I got this error message:
![error](https://github.com/harp-tech/archive-harp-tech.github.io/assets/2701493/99c5244e-3fda-45c8-8295-ed7d16a9415b)

I'll write to eops to see how I can get it fixed.

**Reply by filcarv** [reply]
Hi 👋 OEPS team here.

We've received your clock synchronizer.
We have the replacement ready to ship back. Our team will send you an email to work on the details of the shipment.
